.github/workflows/fmt_build_tests.yml

```diff
@@ -0,0 +1,42 @@
+name: cargo-gccrs CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      main
+
+jobs:
+  fmt-lint-build:
+    runs-on: ubuntu-latest
+    env:
+      RUSTFLAGS: "-D warnings"
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Check the coding style
+        run: |
+          cargo fmt -- --check
+
+      - name: Check for successful compilation
+        run: cargo check
+
+      - name: Check for clippy warnings
+        run: cargo clippy
+
+  unit-and-integration-tests:
+    needs: fmt-lint-build
+    runs-on: ubuntu-latest
+    container:
+      image: philberty/gccrs
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Run unit and functional tests
+        run: |
+          apt install ca-certificates -y
+          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+          . $HOME/.cargo/env
+          cargo test
```

Comment:
Maybe split this into an install rustup and a run tests step?

Reply:
The only issue with this is that I can't get dash to execute the config file where the rust installer added the `source $HOME/.cargo/env` line. We could also install rust using Ubuntu's package manager but I'd rather use the official version

Reply:
I see. It is fine to leave it as is.